index.js

```diff
@@ -0,0 +1,162 @@
+//  console.log(rock.id);
+  moveRock(rock.id);
+}
+
+
+function moveRock(value) {
+//      console.log(value);
+      var rockId = value;
+      var currentTopValue = parseInt(document.getElementById(rockId).style.top.replace('px', ''));
+//      console.log(currentTopValue);
+      function run() {
+        if (currentTopValue !== 380 ) {
+          console.log(currentTopValue);
+          var rockNewValue = currentTopValue += 2;
+          document.getElementById(rockId).style.top = `${rockNewValue}px`;
+          window.requestAnimationFrame(run);
+          if (checkCollision(document.getElementById(rockId))) {
+            endGame();
+          return;
+          }
+        }
+      else {
+          document.getElementById(rockId).remove();
+          ROCKS.pop();
+          return;
+      }
+    }
+  window.requestAnimationFrame(run);
+}
+
+function endGame() {
+  gameInterval = null;
+  //removal all ROCKS from the DOM
+  $('.rock').remove();
+  //removing moveDodger event listener
+  window.removeEventListener('keydown', moveDodger)
+  alert('YOU LOSE!')
+}
+
+function moveDodger(event) {
+
+     switch(event.which) {
+       case 37:
+       event.preventDefault();
+       event.stopPropagation();
+       moveDodgerLeft();
+       break;
+       case 39:
+       event.preventDefault();
+       event.stopPropagation();
+       moveDodgerRight();
+       break;
+       default: return; // exit this handler for other keys
+   }
+
+}
+function moveDodgerLeft() {
+  var leftOriginalNumber = parseInt(DODGER.style.left.replace('px', ''));
+    if (leftOriginalNumber > 0 ) {
+      var leftNewNumber = leftOriginalNumber -= 4;
+      DODGER.style.left = `${leftNewNumber}px`;
+    }
+ }
+
+function moveDodgerRight() {
+   var rightOriginalNumber = parseInt(DODGER.style.left.replace('px', ''));
+     if (rightOriginalNumber <360 ) {
+      var rightNewNumber = rightOriginalNumber += 4;
+      DODGER.style.left = `${rightNewNumber}px`;
+    }
+}
+
+
+/** below causes collisions when left and right are called after each other
+function moveDodgerLeft() {
+  var leftOriginalNumber = parseInt(DODGER.style.left.replace('px', ''));
+  function run(){
+    if (leftOriginalNumber > 0 ) {
+      var leftNewNumber = leftOriginalNumber -= 4;
+      DODGER.style.left = `${leftNewNumber}px`;
+      window.requestAnimationFrame(run);
+    }
+  }
+  window.requestAnimationFrame(run);
+ }
+
+function moveDodgerRight() {
+   var rightOriginalNumber = parseInt(DODGER.style.left.replace('px', ''));
+   function run(){
+     if (rightOriginalNumber <360 ) {
+      var rightNewNumber = rightOriginalNumber += 4;
+      DODGER.style.left = `${rightNewNumber}px`;
+      window.requestAnimationFrame(run);
+    }
+  }
+  window.requestAnimationFrame(run);
+}
+**/
+/**
+ * @param {string} p The position property
+ * @returns {number} The position as an integer (without 'px')
+ */
+function positionToInteger(p) {
+  return parseInt(p.split('px')[0]) || 0
```

Comment:
You always need to pass a `radix` param to `parseInt` to avoid bugs. Additionally, there's no need to get the number portion of the string, as `parseInt` will handle that.